spec/predicates/release.md

```diff
@@ -0,0 +1,98 @@
+# Predicate type: Release
+
+Type URI: https://in-toto.io/attestation/release
+
+Version 0.1
+
+## Purpose
+
+To authoritatively link from a specific release name and version string in a registry, to the artifact names and hashes that make up that release.
+
+## Use Cases
+
+When receiving a new release version, package registries can publish a release attestation covering the artifact names and hashes that make up that release. This allows consumers of that release version to ensure the artifacts they are consuming have not been tampered with.
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
```

Comment:
Could you clarify the semantics around multiple attestations for a single release?

- Is a registry allowed to issue multiple attestations for a given `{purl, name}` pair? (I'm guessing that is legal if and only if the registry's semantics have mutable releases?)
- Are you allowed to issue separate attestations per file (rather than one attestation with all files)? If so, is one form recommended over the other?

Reply:
Ah yes, I should have mentioned immutability before! I've added clarification to structuring the subject for mutable and immutable releases below.